src/test/ui/existential_types/generic_duplicate_param_use.rs

```diff
@@ -1,8 +1,10 @@
+// compile-pass
 #![feature(existential_type)]
 
 fn main() {}
 
-existential type Two<T, U>: 'static; //~ ERROR type parameter `U` is unused
+// test that unused generic parameters are ok
+existential type Two<T, U>: 'static;
 
 fn one<T: 'static>(t: T) -> Two<T, T> {
```

Comment:
Under the RFC, this function doesn't qualify as a defining use because multiple mappings from `Two<T, T>` to type `T` could exist. This is especially clear in this example, where (AFAICT) there's no reason that `U` is unused, but `T` isn't unused-- it could be that `T` is unused but `U` is used instead (i.e. `type Two<T, U> = T;` vs `type Two<T, U> = U;`).

In order for something to qualify as a defining use, all type parameters must be able to vary independently, e.g. `fn one<T, U>(..) -> Two<T, U>`. The rules for this are actually probably similar-ish to the ones that @nikomatsakis had in mind for "always applicable impls" for specialization.

Reply:
AFAICT it's not possible for this function to be a defining use and to anonymously agree with a separate function `fn t_first<T, U>(x: T) -> Two<T, U> { x }` *or* a separate function `fn t_second<T, U>(x: T) -> Two<U, T> { x }`. Obviously both can't exist at the same time, but *one* of them could exist and be a defining use, but just looking at `fn one<T>(t: T) -> Two<T, T> { t }` we don't know which should be allowed.

Reply:
I'm iterating over all the functions anyway, so I can implement an order-independent algorithm. The question is whether that should be done. It sure feels surprising